Sources/ProjectDescription/Config.swift

```diff
@@ -75,14 +60,34 @@ public struct Config: Codable, Equatable, Sendable {
         generationOptions: GenerationOptions = .options(),
         installOptions: InstallOptions = .options()
     ) {
-        self.compatibleXcodeVersions = compatibleXcodeVersions
-        self.plugins = plugins
-        self.generationOptions = generationOptions
-        self.installOptions = installOptions
-        self.cloud = cloud
+        var fullHandle = fullHandle
+        var url = url
+        var generationOptions = generationOptions
+        if let cloud {
+            fullHandle = cloud.projectId
+            url = cloud.url
+            generationOptions.optionalAuthentication = cloud.options.contains(.optional)
+        }
```

Comment:
```suggestion
        self.fullHandle = fullHandle ?? cloud.projectId
        self.url = url ?? cloud.url
        var generationOptions = generationOptions
        if let cloud {
            generationOptions.optionalAuthentication = cloud.options.contains(.optional)
        }
```

I'm trying to avoid `var xxx` as alternatives are imho clearer. When using a `var` you need to scan the rest of the method to see the places where it's modified. It's relatively simple in this case, but in other case it's often not.